Changelog 0.12.10 (#4775)
dagster-io/dagster: CHANGES.md

# Changelog

# 0.12.10

### Community Contributions

> - [helm] The Kubernetes run launcher image pull policy is now configurable in a separate field (thanks [@yamrzou](https://github.com/yamrzou)!).

Reviewer: ```suggestion
- [helm] The `KubernetesRunLauncher` image pull policy is now configurable in a separate field (thanks [@yamrzou](https://github.com/yamrzou)!).
```

- The `dagster-github` package is now usable for GitHub Enterprise users (thanks [@metinsenturk](https://github.com/metinsenturk)!) A hostname can now be provided via config to the dagster-github resource with the key `github_hostname`:

```
execute_pipeline(
      github_pipeline, {'resources': {'github': {'config': {
           "github_app_id": os.getenv('GITHUB_APP_ID'),
           "github_app_private_rsa_key": os.getenv('GITHUB_PRIVATE_KEY'),
           "github_installation_id": os.getenv('GITHUB_INSTALLATION_ID'),
           "github_hostname": os.getenv('GITHUB_HOSTNAME'),
      }}}}
)
```

### New

- Added a database index over the event log to improve the performance of `pipeline_failure_sensor` and `run_status_sensor` queries. To take advantage of these performance gains, run a schema migration with the CLI command: `dagster instance migrate`.

### Bugfixes

- Performance improvements have been made to allow dagit to more gracefully load a run that has a large number of events.

> - Fixed an issue where DockerRunLauncher would raise an exception when no networks were specified in its configuration.

Reviewer: ```suggestion
- Fixed an issue where `DockerRunLauncher` would raise an exception when no networks were specified in its configuration.
```

### Breaking Changes

- `dagster-slack` has migrated off of deprecated [`slackclient` (deprecated)](http://slackclient/) and now uses `[slack_sdk](https://slack.dev/python-slack-sdk/v3-migration/)`.

### Experimental

> - `OpDefinition`, the replacement for `SolidDefinition` which is the type produced by the `@op` decorator, is now part of the public API

Reviewer: ```suggestion
- `OpDefinition`, the replacement for `SolidDefinition` which is the type produced by the `@op` decorator, is now part of the public API.
```
